dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report.test.ts

```diff
@@ -0,0 +1,53 @@
+import httpMocks from 'node-mocks-http';
+import { UTILISATION_REPORT_RECONCILIATION_STATUS, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { UtilisationReportRepo } from '../../../repositories/utilisation-reports-repo';
+import { GetUtilisationReportResponse } from '../../../types/utilisation-reports';
+import { GetUtilisationReportByIdRequest, getUtilisationReportById } from './get-utilisation-report.controller';
+
+jest.mock('../../../repositories/utilisation-reports-repo');
+
+jest.mock('../../../repositories/utilisation-reports-repo/utilisation-report-sql.repo');
+
+console.error = jest.fn();
+
+describe('getUtilisationReport', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const getHttpMocks = () =>
+    httpMocks.createMocks<GetUtilisationReportByIdRequest>({
+      params: { id: '2023' },
+    });
+
+  it('sends a 200 and maps entity to response', async () => {
+    // Arrange
+    const mockUtilisationReport = UtilisationReportEntityMockBuilder.forStatus(UTILISATION_REPORT_RECONCILIATION_STATUS.PENDING_RECONCILIATION).build();
+    const findOneByMock = jest.fn().mockResolvedValue(mockUtilisationReport);
+    jest.spyOn(UtilisationReportRepo, 'findOneBy').mockImplementation(findOneByMock);
+
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await getUtilisationReportById(req, res);
+
+    // Assert
+    expect(res.statusCode).toEqual(200);
+    // eslint-disable-next-line no-underscore-dangle
+    const responseData = res._getData() as GetUtilisationReportResponse;
+    expect(responseData.id).toEqual(mockUtilisationReport.id);
```

Comment:
can we do anything interesting like:
expect(expect.ObjectContaining(responseData)).toEqual(mockUtilisationReport)? Assuming this is just mockUtilisationReport has more properties than response data. or do you think that makes it harder to maintian in the future?

Reply:
The other way around like so: `expect(mockUtilisationReport).toEqual(expect.objectContaining(responseData));` gets close to working, however it doesn't deal with the nested objects also just being subsets, which is the case for the azure file info, so I think we would have to do something even more complex at which point I quite like the simplicity of each of these lines even though it is more lines (which one might forget to add to in the future if adding fields, which is a concern I guess but I don't think it is too big of one).